core/controllers/profile_test.py

```diff
@@ -236,9 +238,16 @@ def test_get_profile_picture_valid_username(self):
         response_dict = self.get_json(
             '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)
         )
-        # Although the user has a valid username, they have not yet supplied
-        # a profile picture.
-        self.assertIsNone(
+        self.assertIsNotNone(
+            response_dict['profile_picture_data_url_for_username'])
+
+    def test_get_signup_gravatar(self):
+        self.signup(self.EMAIL, self.USERNAME)
+        response_dict = self.get_json(
+            '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)
+        )
+        self.assertEqual(
+            self.get_gravatar_from_url(self.GRAVATAR_URL),
```

Comment:
This is not testing anything -- you're calling a function in the test setup, so if anything goes wrong with fetch_gravatar(), the test would not detect it. This makes the test not useful.

Instead, it's better, in the local unit tests for fetch_gravatar(), to swap out urlfetch.fetch() with a custom method that returns something specific. Among other things, this also makes it possible for the test to run without a network connection.